pkg/plugins/crypto/impl/crypto.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package impl provides the default implementation of pkg/plugins/crypto.Crypto interface
+//
+// It uses github.com/tink/go crypto primitives
+package impl
+
+import (
+	"errors"
+
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/core/primitiveset"
+	"github.com/google/tink/go/keyset"
+	aeadsubtle "github.com/google/tink/go/subtle/aead"
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/hyperledger/aries-framework-go/pkg/plugins/crypto"
+)
+
+// DefaultCrypto is the default Crypto implementation
+type DefaultCrypto struct {
+}
+
+// Assert that DefaultCrypto implements the AriesCrypto interface.
+var _ crypto.AriesCrypto = (*DefaultCrypto)(nil)
+
+// Encrypt will encrypt msg using the implementation's corresponding encryption key
+func (t *DefaultCrypto) Encrypt(msg, aad []byte, kh *keyset.Handle) ([]byte, []byte, error) {
+	a, err := aead.New(kh)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ct, err := a.Encrypt(msg, aad)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ps, err := kh.Primitives()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ivSize := getNonceSize(ps)
+	cipherText := ct[len(ps.Primary.Prefix)+ivSize:]
+	nonce := ct[len(ps.Primary.Prefix) : ivSize-1]
+
+	return cipherText, nonce, nil
+}
+
+func getNonceSize(ps *primitiveset.PrimitiveSet) int {
```

Comment:
this method doesn't seem to `get` anything.

Reply:
it returns the nonce size of the primitives we want to support, either AESGCM nonce size or Xchacha20Poly1305 nonce size

Reply:
right - this method should not have a get prefix.

Reply:
removed `get` prefix